src/marker.rs

```diff
@@ -882,6 +866,29 @@ impl<'py> Python<'py> {
     }
 }
 
+impl<'unbound> Python<'unbound> {
+    /// Unsafely creates a Python token with an unbounded lifetime.
+    ///
+    /// Many of PyO3 APIs use `Python<'_>` as proof that the GIL is held, but this function can be
+    /// used to call them unsafely.
+    ///
+    /// # Safety
+    ///
+    /// - This token and any borrowed Python references derived from it can only be safely used
+    /// whilst the currently executing thread is actually holding the GIL.
```

Comment:
Re: threads, I was thinking about this:

```rust
fn foo(py: Python<'_>){
    let handle = std::thread::spawn(|| unsafe {
        let py = Python::assume_gil_acquired();
        // call some pyo3 apis
    });
    
    handle.join().unwrap();
}
```

Is this legal?

Reply:
As `spawn` and `join` are synchronization points, I think this is legal as long as there is absolutely nothing happening in between `spawn` and `join` which could prevent the `join` from happening, i.e. the thread must be scoped.

However, I would suggest not including this example in the documentation as it opens more questions than it answers IMHO and I think the current wording is the right thing to do in almost all of the cases. When someone has to handle a case where this is not so (as in your example), they should be sure enough to figure it out. Everybody is probably just confused by watering down the statement about the current thread.

Reply:
To clarify, the way I worded it would declare that it's not allowed.